crates/hir_def/src/body.rs

```diff
@@ -121,11 +121,8 @@ impl Expander {
                 err.get_or_insert(e);
             })?;
         let call_id = match call_id {
-            Some(it) => it,
-            None => {
-                if err.is_none() {
-                    log::warn!("no error despite `as_call_id_with_errors` returning `None`");
```

Comment:
I'm opting to remove this check now that it's using the ErrorEmitted pattern which cannot be constructed without writing to the sink.